internal/core/validations/validations.go

```diff
@@ -78,3 +79,31 @@ func IsVersionValid(version string) bool {
 	_, err := semver.NewVersion(version)
 	return err == nil
 }
+
+// IsNameValidOnKube check if name is valid for kubernetes resources (RFC 1123)
+// Since regexp does not support lookbehind, we test directly the last character before
+// matching the string name.
+func IsNameValidOnKube(name string) bool {
```

Comment:
@caiordjesus I think that we can assume some pattern here like the previous validations which use _**is/are**_ as prefix,  _**field**_ and then _**action**_ as suffix

examples:
Is MaxSurge _**Valid**_
Is ImagePullPolicy _**Supported**_
Is Protocol _**Supported**_
Is Version _**Valid**_
Is Name _**Valid**_ OnKube

WDYT about use IsKubeResourceNameValid?